File: packages/python/prompt_sanitizer/entities.py
from enum import Enum


class EntityType(str, Enum):
    """All PII / sensitive entity types that prompt-sanitizer can detect."""

    # ── Personal identifiers ────────────────────────────────────────────────
    PERSON = "PERSON"
    EMAIL = "EMAIL"
    PHONE = "PHONE"
    DATE_OF_BIRTH = "DATE_OF_BIRTH"
    AGE = "AGE"

    # ── Location ────────────────────────────────────────────────────────────
    ADDRESS = "ADDRESS"
    ZIP_CODE = "ZIP_CODE"

    # ── Financial ───────────────────────────────────────────────────────────
    CREDIT_CARD = "CREDIT_CARD"
    IBAN = "IBAN"
    BANK_ACCOUNT = "BANK_ACCOUNT"
    CRYPTO_ADDRESS = "CRYPTO_ADDRESS"

    # ── Government IDs ──────────────────────────────────────────────────────
    SSN = "SSN"
    PASSPORT = "PASSPORT"
    DRIVING_LICENSE = "DRIVING_LICENSE"

    # ── Network / Digital ───────────────────────────────────────────────────
    IP_ADDRESS = "IP_ADDRESS"
    MAC_ADDRESS = "MAC_ADDRESS"
    URL = "URL"

    # ── Secrets & credentials ────────────────────────────────────────────────
    API_KEY = "API_KEY"
    JWT = "JWT"
    BEARER_TOKEN = "BEARER_TOKEN"
    AWS_ACCESS_KEY = "AWS_ACCESS_KEY"
    AWS_SECRET_KEY = "AWS_SECRET_KEY"
    PRIVATE_KEY = "PRIVATE_KEY"
    DB_CONNECTION = "DB_CONNECTION"

    # ── Temporal ────────────────────────────────────────────────────────────
    DATE = "DATE"

    # ── User-defined ────────────────────────────────────────────────────────
    CUSTOM = "CUSTOM"

File: packages/ruby/lib/prompt_sanitizer/entities.rb
# frozen_string_literal: true

module PromptSanitizer
  # All PII / sensitive entity types the gem can detect.
  # Values are symbols — mirror the Python/JS string enum names.
  module EntityType
    # ── Personal identifiers ──────────────────────────────────────────────
    PERSON        = :person
    EMAIL         = :email
    PHONE         = :phone
    DATE_OF_BIRTH = :date_of_birth
    AGE           = :age

    # ── Location ──────────────────────────────────────────────────────────
    ADDRESS       = :address
    ZIP_CODE      = :zip_code

    # ── Financial ─────────────────────────────────────────────────────────
    CREDIT_CARD   = :credit_card
    IBAN          = :iban
    BANK_ACCOUNT  = :bank_account
    CRYPTO_ADDRESS = :crypto_address

    # ── Government IDs ────────────────────────────────────────────────────
    SSN             = :ssn
    PASSPORT        = :passport
    DRIVING_LICENSE = :driving_license

    # ── Network / Digital ─────────────────────────────────────────────────
    IP_ADDRESS  = :ip_address
    MAC_ADDRESS = :mac_address
    URL         = :url

    # ── Secrets & credentials ─────────────────────────────────────────────
    API_KEY        = :api_key
    JWT            = :jwt
    BEARER_TOKEN   = :bearer_token
    AWS_ACCESS_KEY = :aws_access_key
    AWS_SECRET_KEY = :aws_secret_key
    PRIVATE_KEY    = :private_key
    DB_CONNECTION  = :db_connection

    # ── Temporal ──────────────────────────────────────────────────────────
    DATE = :date

    # ── NER-only (detected by SMART / FULL mode) ──────────────────────────
    ORGANIZATION = :organization
    LOCATION     = :location
    MISC         = :misc

    # ── User-defined ──────────────────────────────────────────────────────
    CUSTOM = :custom

    ALL = constants(false).map { |c| const_get(c) }.freeze
  end
end
